apps/api/src/mail/emails/workspace-invitation.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react'
+import {
+  Body,
+  Button,
+  Container,
+  Head,
+  Heading,
+  Html,
+  Link,
+  Preview,
+  Section,
+  Text
+} from '@react-email/components'
+import dayjs from 'dayjs'
+import {
+  container,
+  content,
+  ctaButton,
+  footer,
+  footerText,
+  h1,
+  link,
+  main,
+  text,
+  workspaceDetails,
+  workspaceInfo
+} from '../styles/common-styles'
+
+interface WorkspaceInvitationEmailProps {
+  projectName: string
+  projectUrl: string
+  invitedBy: string
+  invitedOn: string
+  forRegisteredUser: boolean
+}
+
+export const WorkspaceInvitationEmail = ({
+  projectName,
+  projectUrl,
+  invitedBy,
+  invitedOn,
+  forRegisteredUser
+}: WorkspaceInvitationEmailProps) => {
+  const formattedInvitedOnDate = dayjs(invitedOn).format(
+    'ddd, MMM D, YYYY h:mm A'
+  )
+
+  return (
+    <Html>
+      <Head />
+      <Preview>
+        {forRegisteredUser
+          ? 'Welcome Back! Join Your Workspace'
+          : 'You are Invited to Join the Workspace'}
+      </Preview>
+      <Body style={main}>
+        <Container style={container}>
```

Comment:
I'm pretty sure we can re-use the frames in here. We can convert this into a reusable component where in we can pass the body, the rest would be pre set.